pkg/config/runtime_config.go

```diff
@@ -114,37 +117,42 @@ func BuildRestConfig(rtCfg RuntimeConfig) (*rest.Config, error) {
 func BuildRuntimeOptions(rtCfg RuntimeConfig, scheme *runtime.Scheme) ctrl.Options {
 	return ctrl.Options{
 		Scheme:                     scheme,
-		Port:                       rtCfg.WebhookBindPort,
-		CertDir:                    rtCfg.WebhookCertDir,
-		MetricsBindAddress:         rtCfg.MetricsBindAddress,
 		HealthProbeBindAddress:     rtCfg.HealthProbeBindAddress,
 		LeaderElection:             rtCfg.EnableLeaderElection,
-		LeaderElectionResourceLock: resourcelock.ConfigMapsLeasesResourceLock,
+		LeaderElectionResourceLock: resourcelock.LeasesResourceLock, // resourcelock.ConfigMapsLeasesResourceLock,
 		LeaderElectionID:           rtCfg.LeaderElectionID,
 		LeaderElectionNamespace:    rtCfg.LeaderElectionNamespace,
-		Namespace:                  rtCfg.WatchNamespace,
```

Comment:
this broken original functionality. When `rtCfg.WatchNamespace` is not empty, it's supposed to only watch Objects(ingress/service/tgb/pods) in specific namespaces.
in this change, i only saw it's passed to pods informer, but not to other the informers.
We need to test and find a alternative in updated controller runtime. it seems to be https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/cache/cache.go#L192, but need code reading and tests to find out

Reply:
I am going to do some additional research on this, and will answer asap.

Reply:
I believe you are correct on this. I have added `DefaultNamespaces` to the cache options.